impl/client/src/main/javascript/web/pentaho/action/Base.js

```diff
@@ -80,51 +80,96 @@ define([
      * Gets the type of action.
      *
      * @name type
-     * @memberOf pentaho.action.Abstract#
+     * @memberOf pentaho.action.Base#
      * @type {string}
+     * @readonly
      *
      * @abstract
      */
     get type() {
     },
 
+    /**
+     * Gets the event name of the action.
```

Comment:
When `@name` is used, a doclet independent of any surrounding code is created. So, you may remove the implementation. Idem for all other abstract doclets in this file.